<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1366</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Groupbox */

QGroupBox {
    border: 1px solid #DDD;
    border-radius: 9px;
    margin-top: 9px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="OptionsGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Options</string>
      </property>
      <widget class="QSpinBox" name="SubdivSteps">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>90</y>
         <width>181</width>
         <height>22</height>
        </rect>
       </property>
       <property name="maximum">
        <number>8</number>
       </property>
      </widget>
      <widget class="QLabel" name="SubdivLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>181</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Loop subdivision steps</string>
       </property>
      </widget>
      <widget class="QPushButton" name="LoadOBJ">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>181</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Load OBJ file</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="displayGB">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>220</y>
         <width>191</width>
         <height>291</height>
        </rect>
       </property>
       <property name="title">
        <string>Display</string>
       </property>
       <widget class="QSlider" name="reflDensitySlider">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>190</y>
          <width>160</width>
          <height>18</height>
         </rect>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
       <widget class="QComboBox" name="shaderComboBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>110</y>
          <width>151</width>
          <height>33</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>Normal Shading</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Gaussian Curvature</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Flat Shading</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Reflection Lines</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>170</y>
          <width>161</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Reflection Line Density</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="curScalingSB">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>220</y>
          <width>45</width>
          <height>27</height>
         </rect>
        </property>
        <property name="minimum">
         <number>-5</number>
        </property>
        <property name="maximum">
         <number>5</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>90</y>
          <width>111</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Shading Options</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_3">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>220</y>
          <width>111</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Curvature Scaling</string>
        </property>
       </widget>
       <widget class="QPushButton" name="rotResetButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>260</y>
          <width>151</width>
          <height>27</height>
         </rect>
        </property>
        <property name="text">
         <string>Reset Rotation</string>
        </property>
       </widget>
       <widget class="QComboBox" name="drawModeComboBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>151</width>
          <height>33</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>Wireframe</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Surface</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Point Cloud</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_4">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>30</y>
          <width>131</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Draw mode</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="MainView" name="MainDisplay"/>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
